scripts.js

```diff
@@ -0,0 +1,241 @@
+};
+
+console.log(checkIfPropertyIsDefined(user, 'firstName')); // true
+console.log(checkIfPropertyIsDefined(user, 'height')); // false
+console.log(checkIfPropertyIsDefined(user, 'age')); // false
+
+
+// 2
+const salaries = {
+  john: 50,
+  adam: 100,
+  katie: 75
+}
+const salarySum = salaries.john + salaries.adam + salaries.katie;
+
+
+// 3
+function isObject(value) {
+  return typeof value === 'object' && value !== null;
+}
+
+console.log(isObject({})); // true
+console.log(isObject(10)); // false
+console.log(isObject(null)); // false
+
+// 4
+function checkIfPropertyIsDefined(object, propertyName) {
+  return object[propertyName] !== undefined;
+}
+
+function removeProperty(object, propertyName) {
+
+  if (checkIfPropertyIsDefined(object, propertyName)) {
+    delete object[propertyName];
+    return true;
+  }
+  return false;
+}
+
+const userObject = {
+  name: 'John'
+}
+
+removeProperty(userObject, 'name'); // true
+removeProperty(userObject, 'name'); // false
+removeProperty(userObject, 'weight'); // false
+
+console.log(userObject); // {}
+
+
+// 5
+function checkIfUsersHaveTheSameName(firstObject, secondObject) {
+  return firstObject['firstName'] === secondObject['firstName'];
+}
+
+const firstJohn = {
+  firstName: 'John',
+  lastName: 'Smith'
+}
+
+const secondJohn = {
+  firstName: 'John',
+  lastName: 'Smith'
+}
+
+const adam = {
+  firstName: 'Adam',
+  lastName: 'Johnson'
+}
+
+console.log(checkIfUsersHaveTheSameName(firstJohn, secondJohn)); // true
+console.log(checkIfUsersHaveTheSameName(firstJohn, adam)); // false
+
+// 6
+function getVoteCount(votes) {
+  return votes.upvotes - votes.downvotes;
+}
+
+console.log(getVoteCount({ upvotes: 10, downvotes: 5 })); // 5
+console.log(getVoteCount({ upvotes: 75, downvotes: 90 })); // -15
+console.log(getVoteCount({ upvotes: 50, downvotes: 50 })); // 0
+
+// 7
+function getCubeVolume(cubeDimensions) {
+  return cubeDimensions.width * cubeDimensions.height * cubeDimensions.length;
+}
+
+console.log(getCubeVolume({ width: 10, length: 5, height: 2 })); // 100
+console.log(getCubeVolume({ width: 100, length: 500, height: 0 })); // 0
+console.log(getCubeVolume({ width: 15, length: 2, height: 5 })); // 150
+
+// 8
+function getCityInformation(cityObject) {
+  return `${cityObject.name} is in ${cityObject.country} and has an area of ${cityObject.areaInKilometers} square kilometers`;
+}
+
+console.log(getCityInformation({
+  name: 'Warsaw',
+  country: 'Poland',
+  areaInKilometers: 517
+})); // Warsaw is in Poland and has an area of 517 square kilometers
+
+// 9
+function getUserCopy(user) {
+  return userData = {
+    firstName: user['firstName'],
+    lastName: user['lastName'],
+  };
+}
+
+const john = {
+  firstName: 'John',
+  lastName: 'Smith'
+}
+
+const newUser = getUserCopy(john);
+
+console.log(newUser.firstName); // John
+console.log(newUser.lastName); // Smith
+console.log(newUser === john); // false <-- this is crucial
+
+//10
+function getAnimalObject(animalObject) {
```

Comment:
The name `getAnimalObject` suggest that this function returns an object, which is untrue